coffee_core/src/coffee.rs

```diff
@@ -272,11 +273,31 @@ impl PluginManager for CoffeeManager {
         })
     }
 
-    async fn upgrade(&mut self, _: &[&str]) -> Result<(), CoffeeError> {
+    async fn upgrade(&mut self, repo: &str, all: bool) -> Result<CoffeeUpgrade, CoffeeError> {
```

Comment:
```suggestion
    async fn upgrade(&mut self, repo: Option<String>) -> Result<CoffeeUpgrade, CoffeeError> {
```

Maybe this is a good signature?

Reply:
Yes this makes more sense.
I will use this approach when I work on adding support for `all` flag